# Lab 2.03 - Univariate Analysis


In [2]:
# Importing the necessary packages
import numpy as np                                  # "Scientific computing"
import scipy.stats as stats                         # Statistical tests

import pandas as pd                                 # Data Frame
from pandas.api.types import CategoricalDtype

import matplotlib.pyplot as plt                     # Basic visualisation
from statsmodels.graphics.mosaicplot import mosaic  # Mosaic diagram
import seaborn as sns                               # Advanced data visualisation

## Exercise 3 - Kitten litters

A vet keeps a record of the number of kittens in litters produced by cats in his practice. The code block below contains the data as a frequency table.

- First put the data into one list of individual observations: [1, 1, 2, 2, ... 5, 5, 6, 6, 6, 6, 7, 7, 8]  
- Visualise the data set using all suitable chart types that you know of.
- Calculate:
    - the median size of a litter
    - the mean size of a litter
    - the lower quartile
    - the upper quartile

In [6]:
l = []
kittens = pd.DataFrame.from_dict({
    'Number in litter': [1, 2, 3,  4, 5, 6, 7, 8],
    'Frequency':        [2, 4, 7, 11, 8, 4, 2, 1]})

for n,f in zip(kittens['Number in litter'],kittens['Frequency']):
    l.extend([n]*f)
l = np.array(l)
print(l.mean())
print(np.percentile(l, 25))
print(np.percentile(l, 75))

4.128205128205129
3.0
5.0


### Some answers:

- Median: 4.0
- Mean: 4.128205128205129
- Lower Quartile: 3.0
- Upper Quartile: 5.0

Source: Complete Probability & Statistics 1 for Cambridge International As & A Level